homeassistant/components/lcn.py

```diff
@@ -98,8 +101,8 @@ def is_address(value):
     vol.Required(CONF_USERNAME): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
     vol.Optional(CONF_SK_NUM_TRIES, default=3): cv.positive_int,
-    vol.Optional(CONF_DIM_MODE, default='steps50'): vol.All(vol.In(DIM_MODES),
-                                                            vol.Upper),
+    vol.Optional(CONF_DIM_MODE, default='steps50'): vol.All(vol.Upper,
```

Comment:
Does this make any difference ? If it doesn't please revert back.

Reply:
I think it's not obious at first sight but I try to explain.
I assume the following facts:

1. All formatting should be done within the validators.
2. `configuration.yaml` uses lowercase.
3. Underlying library strongly depends on enums which use uppercase (since Python3.6 docs say to do so).

I directly want to use the values from `configuration.yaml` as enum keys, so there has to be somewhere this case conversion.
The changes in this PR didn't make any difference until I introduced the conditional if clause in the light's `async_setup_platform` function (I commented the position below).
There is a comparison if `config[OUTPUT]` is in `OUTPUT_PORTS` (or `RELAY_PORTS).` Without changing the constant lists to uppercase and exchange the position of the validators, I would have to use `config[CONF_OUTPUT].lower()` which is inconsistent with fact 1.

Reply:
In this validation you essentially have the same vol.All() you just changed the order...